testutil/integration/evmos/utils/erc20.go

```diff
@@ -0,0 +1,114 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package utils
+
+import (
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/evmos/evmos/v15/testutil/integration/evmos/factory"
+	"github.com/evmos/evmos/v15/testutil/integration/evmos/network"
+	erc20types "github.com/evmos/evmos/v15/x/erc20/types"
+)
+
+// ERC20RegistrationData is the necessary data to provide in order to register an ERC20 token.
+type ERC20RegistrationData struct {
+	// Address is the address of the ERC20 token.
+	Address common.Address
+	// Denom is the ERC20 token denom.
+	Denom string
```

Comment:
I would probably leave this as is, it would be kind of double wrapped for no real reason, because these fields are not passed on bundled together anywhere in the function 🙏 this function is also called at a point where no token pair was created yet, so it could be kind of misleading